feat (CIS) - Managed Rulesets - Do Not merge (#5238)
IBM-Cloud/terraform-provider-ibm: website/docs/r/cis_rulesets_rule.html.markdown

- `cis_id` - (Required, String) The ID of the CIS service instance.
- `domain_id` - (Optional, String) The Domain/Zone ID of the CIS service instance. If domain_id is provided request will be made at zone/domain level else request will be made at instance level.
- `ruleset_id` - (Required, String) Id of the ruleset inside which rules will be created, updated or deleted.
- `rules` (Optional, list) Rules which are required to be added/modified.
  
  Nested scheme of `rules`
    - `action` (String). If we are deploying a rule then action is required. `execute` action is used for deploying the ruleset. If we are updating the rule we then action is optional.
    - `description` (Optional, String) Description of the rule.
    - `enable` (Optional, Boolean) Enables/Disables the rule.
    - `expression` (Optional, String) Expression used by the rule to match the incoming request.
    - `ref` (Optional, String) Id of an existing rule. If not provided it is populated by the id if the new created rule.
    - `action_parmeters` (Optional, list) Parameters which are used to modify the rules.
    
      Nested scheme of `action parameters`

>       - `id` (Required, String) Id of the managed ruleset to be deploted.
>       - `overrides` (Optional, list) provides the parameter which are to be overridden.

Reviewer: ```suggestion
      - `id` (Required, String) ID of the managed ruleset to be deployed.
      - `overrides` (Optional, List) Provides the parameters which are to be overridden.
```

        Nested scheme of `overrides`
        - `action` (Optional, String) Action of the rule. Examples: log, block, skip.

>         - `enabled` (Optional, Boolean) Enables/Disables the rule
>         - `rules` (Optional, list) list of details of managed rules to be overridden. These rules are already present in the ruleset which we are deploying or are already deployed

Reviewer: ```suggestion
        - `enabled` (Optional, Boolean) Enables/Disables the rule.
        - `rules` (Optional, List) List of details of managed rules to be overridden. These rules are already present in the ruleset which are deploying or are already deployed.
```

          Nested scheme of `rules`

>           - `id` (Required, String) id of the rule.
>           - `enabled` (Optional, Boolean) Enables/Disables the rule.
>           - `action` (Optional, String) Action of the rule.
>         - `categories` (Optional, list)

Reviewer: ```suggestion
          - `id` (Required, String) ID of the rule.
          - `enabled` (Optional, Boolean) Enables/Disables the rule.
          - `action` (Optional, String) Action of the rule.
        - `categories` (Optional, List)
```

          
          Nested scheme of `categories`
          - `category` (Required, String) Category of the rule.
          - `enabled` (Optional, Boolean) Enables/Disables the rule.
          - `action` (Optional, String) Action of the rule.

>     - `position` (Optional, list). Provides the postion when a new rule is added or updated the position of the current rule. If not provided new rule will be added at the last. You can only use one of the before, after, and index fields at a time.
>       - `index` (Optional, String) Index of the rule to be added. 
>       - `before` (Optional, String) Id of the rule before which new rule will be added. 
>       - `after` (Optional, String) Id of the rule after which new rule will be added.

Reviewer: ```suggestion
    - `position` (Optional, List) Provides the position when a new rule is added or updates the position of the current rule. If not provided the new rule will be added at the end. You can use only one of the before, after, and index fields at a time.
      - `index` (Optional, String) Index of the rule to be added. 
      - `before` (Optional, String) ID of the rule before which the new rule will be added. 
      - `after` (Optional, String) ID of the rule after which the new rule will be added.
```

        

## Attribute reference

> In addition to all argument reference list, you can access the following attribute reference after your resource is created.
> 
> - `rule_id` - (String) Id of the rule. 

Reviewer: ```suggestion
In addition to the argument reference list, you can access the following attribute reference after your resource is created.

- `rule_id` - (String) ID of the rule. 
```

> ## Import

Author: No read API hence mention that import is not possible

Author: keep the import block

> The `ibm_cis_rulesets_rule` resource is imported by using the ID. The ID is formed from the rule ID, the ruleset ID, the domain ID of the domain and the CRN (Cloud Resource Name) concatenated  using a `:` character.

Reviewer: ```suggestion
The `ibm_cis_rulesets_rule` resource is imported by using the ID. The ID is formed from the rule ID, the ruleset ID, the domain ID of the domain and the Cloud Resource Name (CRN) concatenated  using a `:` character.
```

> The domain ID and CRN is located on the **overview** page of the internet services instance of the domain heading of the console, or by using the `ibm cis` command line commands.
> 
> - **Rule ID** is a 32 digit character string of the form: `dcdec3fe0cbe41edac08619503da8de5`.
> 
> - **Ruleset ID** is a 32 digit character string of the form: `489d96f0da6ed76251b475971b097205c`.
> 
> - **Domain ID** is a 32 digit character string of the form: `9caf68812ae9b3f0377fdf986751a78f`.
> 
> - **CRN** is a 120 digit character string of the form: `crn:v1:bluemix:public:internet-svcs:global:a/4ea1882a2d3401ed1e459979941966ea:31fa970d-51d0-4b05-893e-251cba75a7b3::`.

Reviewer: ```suggestion
The domain ID and CRN are located on the **Overview** page of the Internet Services instance of the domain heading of the console, or by using the `ibm cis` CLI commands.

- **Rule ID** is a 32-digit character string of the form: `dcdec3fe0cbe41edac08619503da8de5`.

- **Ruleset ID** is a 32-digit character string of the form: `489d96f0da6ed76251b475971b097205c`.

- **Domain ID** is a 32-digit character string of the form: `9caf68812ae9b3f0377fdf986751a78f`.

- **CRN** is a 120-digit character string of the form: `crn:v1:bluemix:public:internet-svcs:global:a/4ea1882a2d3401ed1e459979941966ea:31fa970d-51d0-4b05-893e-251cba75a7b3::`.
```
